### PR TITLE
Add editorconfig language server

### PR DESCRIPTION
**Pull request by Lilja**
Hi!

I have made a WIP language server for editorconfig.

Can this server configuration be added upsteam?

### CODE REVIEW

**Comment by glepnir** [comment]
it only works for `.editorconfig` file right?

**Comment by Lilja** [comment]
@glepnir Correct

**Comment by glepnir** [comment]
so why we need `.git` ? and does it not support single file mode ?

**Comment by Lilja** [comment]
> so why we need .git

A copy paste artefact. I'll remove it. 

> does it not support single file mode ?

What do you mean with single file mode?

**Comment by glepnir** [comment]
`single_file_support` option that mean this server can work with a single file without root dir.  in rpc protocol is the `rootUri` is current dir `workspaceFolders` nil . send to server

**Comment by Lilja** [comment]
I tried to read what the `:h lspconfig-single-file-support` said about this and I couldn't fully understand it. It's a regular LS with workspace/root dir support. Editorconfig files can be nested, so I would argue that it would need to know its workspace directory.

**Comment by glepnir** [comment]
add `single_file_support` . it will use  cwd as  root send to your server. does your server can accept this ? 
like open nvim in home dir the cwd is home. and open `~/project/.editorconfig` .  the root actual is project but we send cwd.

File: lua/lspconfig/server_configurations/editorconfig.lua [hunk 1]
```diff
@@ -0,0 +1,24 @@
+local util = require 'lspconfig.util'
+
+local root_files = { '.editorconfig' }
```
Comment (glepnir) [inline lua/lspconfig/server_configurations/editorconfig.lua:3]:
```suggestion
```

File: lua/lspconfig/server_configurations/editorconfig.lua [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+return {
+  default_config = {
+    cmd = { 'ecls', '--stdio' },
+    filetypes = { 'editorconfig' },
```
Comment (glepnir) [inline lua/lspconfig/server_configurations/editorconfig.lua:7]:
```suggestion
    filetypes = { 'editorconfig' },
    root_files = { '.editorconfig' },
```

**Comment by Lilja** [comment]
I tried your suggestions locally and it didn't work for my server. Nothing in `:LspLog` either: img

**Comment by glepnir** [comment]
well. I think it would be better support that. because it only work for `.editorconfig` it just a config file

**Comment by justinmk** [comment]
> I have made a WIP language server for editorconfig.

no thanks. configs should not be added here unless they are actually used by many people.

also why does editorconfig need a lsp server?

**Pull request closed without merging**